qa/SidechainTestFramework/sc_test_framework.py

```diff
@@ -48,25 +53,114 @@ def sync_all(self):
+    Parameters:
+     - network: an instance of SCNetworkConfiguration (see sc_boostrap_info.py)
+                    
+    Example: 2 mainchain nodes and 3 sidechain nodes (with default websocket configuration) bootstrapped, respectively, from mainchain node first, first, and third.
+    The JSON representation is only for documentation.
+    {
+        network: {
+            "sc_creation_info":{
+                "mainchain_node": mc_node_1,
+                "sc_id": "id_1"
+                "forward_amout": 200
+                "withdrawal_epoch_length": 1000
+            },
+            [
+                sidechain_1_configuration: {
+                    "mc_connection_info":{
+                        "address": "ws://mc_node_1_hostname:mc_node_1_ws_port"
+                        "connectionTimeout": 100
+                        "reconnectionDelay": 1
+                        "reconnectionMaxAttempts": 1
+                    }
+                },
+                sidechain_2_configuration: {
+                    "mc_connection_info":{
+                        "address": "ws://mc_node_1_hostname:mc_node_1_ws_port"
+                        "connectionTimeout": 100
+                        "reconnectionDelay": 1
+                        "reconnectionMaxAttempts": 1
+                    }
+                
+                },
+                sidechain_3_configuration: {
+                    "mc_connection_info":{
+                        "address": "ws://mc_node_2_hostname:mc_node_2_ws_port"
+                        "connectionTimeout": 100
+                        "reconnectionDelay": 1
+                        "reconnectionMaxAttempts": 1
+                    }
+                }
+            ]
+        }
+    }
+     
+     Output: a map of:
+     - bootstrap information of the sidechain nodes. An instance of SCBootstrapInfo (see sc_boostrap_info.py)    
+    """
+    def bootstrap_sidechain_nodes(self, network=SCNetworkConfiguration):
+        total_number_of_sidechains = len(network.sc_nodes_configuration)
+        sc_creation_info = network.sc_creation_info
+        sc_nodes_bootstrap_info = self.sidechain_creation(sc_creation_info)
+        for i in range(total_number_of_sidechains):
+            sc_node_conf = network.sc_nodes_configuration[i]
+            self.bootstrap_sidechain_node(i, sc_nodes_bootstrap_info, sc_node_conf)
+        return sc_nodes_bootstrap_info
+
+    """
+    Create a sidechain transaction inside a mainchain node.
     
-    def sc_setup_chain(self):
-        genesisData = self.sc_generate_genesis_data() #Should interact with mainchain in order to generate Genesis Data for SC Nodes
-        initialize_sc_chain_clean(self.options.tmpdir, 1, genesisData)
+    Parameters:
+     - sc_creation_info: an instance of SCCreationInfo (see sc_boostrap_info.py)
+     
+     Output: a map of:
+      - an instance of SCBootstrapInfo (see sc_boostrap_info.py)
+    """
+    def sidechain_creation(self, sc_creation_info):
```

Comment:
the same as above. `sidechain_creation` -> `create_sidechain`?

Reply:
Done